op-supervisor/supervisor/backend/processors/l1_processor.go

```diff
@@ -29,18 +33,20 @@ type L1Processor struct {
 	currentNumber uint64
 	tickDuration  time.Duration
 
-	db chainsDB
+	db  chainsDB
+	snc controller
 
 	ctx    context.Context
 	cancel context.CancelFunc
 	wg     sync.WaitGroup
 }
 
-func NewL1Processor(log log.Logger, cdb chainsDB, client L1Source) *L1Processor {
+func NewL1Processor(log log.Logger, cdb chainsDB, snc controller, client L1Source) *L1Processor {
 	ctx, cancel := context.WithCancel(context.Background())
 	return &L1Processor{
 		client:       client,
 		db:           cdb,
+		snc:          snc,
```

Comment:
instead of holding these two components of the backend individually, they should be represented by the backend directly, who can serve the interface of both.